### ISSUE TITLE
Response type shouldn't come from Shiki

### ISSUE THREAD

**Issue by helloimalastair** [withastro/astro#8507]
### Astro Info

```block
Astro                    v3.0.12
Node                     v18.15.0(bun v1.0.0.)
System                   macOS (arm64)
Package Manager          bun
Output                   static
Adapter                  none
Integrations             none
```

### Describe the Bug

When dependencies are installed, the `Response` global comes from Shiki, which doesn't match with the Web API [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) spec.

### What's the expected result?

Astro should override Shiki's Response with the Web API Standard, or allow overriding to a custom type.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/mismatch

Not sure how relevant this is, since afaict Stackblitz doesn't allow you to inspect the types of objects...

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by Princesseuh** [comment]
Shiki declares the type globally which overrides the type of the dom types 🤦‍♀️ I don't even know why Shiki would have a Response type anyway. You would probably be able to workaround this by defining your own `declare global` in your `env.d.ts`, but this is very clunky. 

We're currently exploring some alternatives to Shiki, which would fix this issue

**Comment by bluwy** [comment]
Fixed in #8502
